Snippets/Snippets_3/Logging/UsingLogging.cs

```diff
@@ -8,6 +8,6 @@ public void SomeMethod()
         //your code
         Logger.Debug("Something interesting happened.");
     }
-    static ILog Logger = LogManager.GetLogger("Name");
+    readonly static ILog Logger = LogManager.GetLogger(typeof(ClassUsingLogging)); // Can also use a string
```

Comment:
It does not nor has it ever added any performance improvement. http://stackoverflow.com/a/277117/53158

it does make the variable pointer immutable. but those benefits are mostly void in the context of a private variable. And even the mutability argument is difficult since people often make things readonly that are mutable like dictionaries or lists. The only approach to immutability is either going full immutable types (pulling in nuget dependencies which we cant do) or moving over to F# which would be a full re-write.

reaonly is only a reminder to someone that you should not change the pointer to a variable other than in the constructor. but tbh if you find yourself forgetting the desired usage of a static variable then the class is either too large or too complicated, or both

static has actual perf benefits since it means it is only instantiate once per type.